### PR TITLE
First iteration on getting a spark job test harness working.

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/main/scala/com/adazza/example/spark/job/WordCount.scala [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package com.adazza.example.spark.job
+
+import org.apache.spark.sql.SparkSession
+import scala.concurrent.ExecutionContext
+
+/**
+  * Example Word Count Job that extends SparkJob trait
+  */
+object WordCount extends AdazzaSparkJob {
+  override def run(implicit ec: ExecutionContext,
```
Comment (ghost) [inline src/main/scala/com/adazza/example/spark/job/WordCount.scala]: Why is this here?

File: build.gradle [hunk 2]
```diff
@@ -8,9 +8,26 @@
...
+apply from: "https://raw.githubusercontent.com/adazza/adazza-gradle/$gradleBranch/adazza-spark-job.gradle"
+//apply from: '../adazza-gradle/adazza-spark-job.gradle'
+apply from: "https://raw.githubusercontent.com/adazza/adazza-gradle/$gradleBranch/adazza-dependency-management.gradle"
+//apply from: '../adazza-gradle/adazza-dependency-management.gradle'
+
+dependencies {
+    compile group: 'org.scalatest', name: "scalatest_$scalaVersion"
+    compile group: 'org.apache.spark', name: 'spark-core_2.11'
+    compile group: 'org.apache.spark', name: 'spark-sql_2.11'
+    compile group: 'org.apache.spark', name: 'spark-mllib_2.11'
```
Comment (jon-adazza) [inline build.gradle:21]: do we really need mllib?

File: src/main/scala/com/adazza/example/spark/job/SparkJobTestHarness.scala [hunk 3]
```diff
@@ -0,0 +1,56 @@
+package com.adazza.example.spark.job
+
+import org.apache.spark.sql.SparkSession
+import org.scalatest.{BeforeAndAfterEach, Suite}
+
+/**
+  * Test Harness for running Adazza Spark Jobs locally.
+  */
+trait SparkJobTestHarness extends BeforeAndAfterEach { self: Suite =>
```
Comment (jon-adazza) [inline src/main/scala/com/adazza/example/spark/job/SparkJobTestHarness.scala]: shouldn't this be in src/test/scala rather than src/main/scala?

File: src/main/scala/com/adazza/example/spark/job/WordCount.scala (hunk 1 above)
Comment (jon-adazza) [reply src/main/scala/com/adazza/example/spark/job/WordCount.scala]: Don't know, but given the problems I ran into with running spark jobs in a threaded context, it probably shouldn't be. The "right" way to do parallelism in the cluster world is to invoke the jobs in parallel using an oozie workflow.

File: src/test/scala/com/adazza/example/spark/job/TestWordCount.scala [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    if (d.exists && d.isDirectory) {
+      d.listFiles.filter(_.isFile).toList
+    } else {
+      List.empty[File]
+    }
+  }
+
+  "WordCount" should "write output file to correct location" in {
+    val inputAbsPath = getClass.getResource("/test-files/usdeclar.txt").getPath
+    val outputAbsPath = s"/tmp/spark-test-harness-$uuid8"
```
Comment (jon-adazza) [inline src/test/scala/com/adazza/example/spark/job/TestWordCount.scala:35]: Don't most of these unit test frameworks already create a scratch area for capturing written files? If we use that, then presumably a "gradle clean" will remove any old cruft that got dropped.

File: src/test/scala/com/adazza/example/spark/job/TestWordCount.scala [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    WordCount.run(ExecutionContext.global, sparkSession, Array(input), output, jobArgs, jobContext)
+
+    // scalatest concurrent and time usage
+    eventually(timeout(2 seconds)) {
+      assertResult(true) {
+        getListOfFiles(outputAbsPath).nonEmpty
+      }
+    }
+  }
+}
```
Comment (jon-adazza) [inline src/test/scala/com/adazza/example/spark/job/TestWordCount.scala:49]: should we add a test for output correctness?
